api/server/init.go

```diff
@@ -26,6 +27,18 @@ func init() {
 	viper.SetDefault(EnvDBURL, fmt.Sprintf("sqlite3://%s/data/fn.db", cwd))
 	viper.SetDefault(EnvLOGDBURL, "") // default to just using DB url
 	viper.SetDefault(EnvPort, 8080)
+
+	if runtime.GOOS != "linux" {
+		// This still lets 10-20 sync/async functions execute concurrently assuming a 2GB machine.
+		viper.SetDefault(EnvFnSyncMemory, 1*1024)
+		viper.SetDefault(EnvFnAsyncMemory, 1*1024)
+		viper.SetDefault(EnvFnAsyncLWMemory, 512) // 512MB for async low water mark
+	} else {
+		viper.SetDefault(EnvFnSyncMemory, 0)    // unlimited (fallback to 20% of docker --memory if possible)
+		viper.SetDefault(EnvFnAsyncMemory, 0)   // unlimited (fallback to 80% of docker --memory if possible)
+		viper.SetDefault(EnvFnAsyncLWMemory, 0) // not set (fallback to 20% of async memory)
```

Comment:
since async and sync pools are partitioned, this is to leave additional room for sync to steal from? is there a good reason to expose this to the user? (it seems complex, i'm not big on knobs but that's me, personally)

I can see the appeal of exposing these for people not on linux who don't want to eat our defaults _and_ don't want to run in docker, I guess (I'm also okay with not caring about said people [until the day comes]). I'm just not so keen on exposing any of these as it means we are basically contracted to support them and it seems like we may one day find a better policy than simply chopping things up by ram blocks for sync and async. since docker is a dependency at this point, I think 99% of cases are covered with `docker --memory` for setting max and then just a 'ratio' knob that may be more straightforward than a user having to calculate the ratios to concrete values a priori, but i'm not big on exposing the knob (supporting the knob) for ratio. it would simplify the configuration quite a bit as well and I think with the same outcome. it wouldn't account for the low watermark but I think that this all moves to a point where we calculate 'usable' ram on the fly every time we request a slot and it would be better to just pass over async requests (with no idea of watermark) if after we get them we see we don't have enough ram to run them, in a 2 phase manner, rather than needing to know we have enough to run one before hand (i think these cases will be only really when people run a really small vm to test out fn and try to run a huge image, which has pebkac issues above all else and we can just spew logs about the situation rather than trying to be precise).